src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedThrowSwitchExpressionExceptionMethod.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeGen;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Throws a 'System.Runtime.CompilerServices.SwitchExpressionException' with the given 'unmatchedValue'.
+    /// </summary>
+    internal sealed class SynthesizedThrowSwitchExpressionExceptionMethod : SynthesizedGlobalMethodSymbol
+    {
+        internal SynthesizedThrowSwitchExpressionExceptionMethod(SourceModuleSymbol containingModule, PrivateImplementationDetails privateImplType, TypeSymbol returnType, TypeSymbol paramType)
+            : base(containingModule, privateImplType, returnType, PrivateImplementationDetails.SynthesizedThrowSwitchExpressionExceptionFunctionName)
+        {
+            this.SetParameters(ImmutableArray.Create(SynthesizedParameterSymbol.Create(this, TypeWithAnnotations.Create(paramType), 0, RefKind.None, "unmatchedValue")));
+        }
+
+        internal override void GenerateMethodBody(TypeCompilationState compilationState, BindingDiagnosticBag diagnostics)
+        {
+            SyntheticBoundNodeFactory F = new SyntheticBoundNodeFactory(this, this.GetNonNullSyntaxNode(), compilationState, diagnostics);
+            F.CurrentFunction = this;
+
+            try
+            {
+                ParameterSymbol unmatchedValue = this.Parameters[0];
+
+                //throw new SwitchExpressionException((object)unmatchedValue);
```

Comment:
I wasn't sure why the pseudocode includes a cast to object here. It looks like we assume that this method parameter type is the same as the constructor parameter type and don't do a conversion. I'm fine with that, just checking.

Reply:
This is [an existing behavior](https://github.com/dotnet/roslyn/blob/b8bd69824fa1f4190cc0a24121828524ec5488ef/src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_SwitchExpression.cs#L130), which if I understand correctly makes sure we call into the correct `SwitchExpressionException` constructor overload.

Reply:
```cs
// we cast to object here
ThrowSwitchExpressionException((object)arg);

void ThrowSwitchExpressionException(object param)
{
    // comment implies another cast to 'object' is included here, but there isn't such a cast
    throw new SwitchExpressionException(param);
}
```

It's perhaps more clear to remove the cast from this comment here, but I also don't have a strong opinion.
```suggestion
                //throw new SwitchExpressionException(unmatchedValue);
```